api/commands/query.ts

```diff
@@ -44,7 +44,7 @@ export async function compile(
   compileConfig?: dataform.ICompileConfig
 ): Promise<string> {
   // If there is no project directory, no need to compile the script.
-  if (!compileConfig || !compileConfig.projectDir) {
+  if (!compileConfig || !compileConfig.projectDir || query === "") {
```

Comment:
i have a strong preference for fixing the bug as close to its source as possible; that must be somewhere inside `core` or inside the `returnOverride` code below. but let's please first figure out what is actually happening